safehttp/plugins/xsrf/xsrf_test.go

```diff
@@ -163,87 +91,204 @@ func TestXSRFTokenMultipart(t *testing.T) {
 			wantBody: "Forbidden\n",
 		},
 	}
-	for _, test := range tests {
-		i := xsrf.Interceptor{AppKey: "xsrf", Identifier: userIdentifier{}}
-		tok := xsrftoken.Generate("xsrf", test.userID, test.host+test.path)
-		b := "--123\r\n" +
-			"Content-Disposition: form-data; name=\"xsrf-token\"\r\n" +
-			"\r\n" +
-			tok + "\r\n" +
-			"--123--\r\n"
-		req := safehttptest.NewRequest("POST", "http://foo.com/pizza", strings.NewReader(b))
-		req.Header.Set("Content-Type", `multipart/form-data; boundary="123"`)
+)
 
-		rec := safehttptest.NewResponseRecorder()
-		i.Before(rec.ResponseWriter, req)
+func TestTokenPost(t *testing.T) {
+	for _, method := range stateChangingMethods {
+		for _, test := range formTokenTests {
+			t.Run(method+test.name, func(t *testing.T) {
+				i := xsrf.Interceptor{AppKey: "xsrf", Identifier: userIdentifier{}}
+				tok := xsrftoken.Generate("xsrf", test.userID, test.actionID)
+				req := safehttptest.NewRequest(method, "https://foo.com/pizza", strings.NewReader(xsrf.TokenKey+"="+tok))
+				req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+				rec := safehttptest.NewResponseRecorder()
 
-		if rec.Status() != test.wantStatus {
-			t.Errorf("response status: got %v, want %v", rec.Status(), test.wantStatus)
-		}
-		if diff := cmp.Diff(test.wantHeader, map[string][]string(rec.Header())); diff != "" {
-			t.Errorf("rw.header mismatch (-want +got):\n%s", diff)
+				i.Before(rec.ResponseWriter, req)
+
+				if rec.Status() != test.wantStatus {
+					t.Errorf("response status: got %v, want %v", rec.Status(), test.wantStatus)
+				}
```

Comment:
It's probably good to have a `got` here since you are making a function call. So that we are 100% sure that the thing being compared is the same as the thing being printed.

The error message could also start with `"rec.Status(): ..."`